plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TimestampCoercer.java

```diff
@@ -97,4 +101,52 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
                             toType));
         }
     }
+
+    public static class VarcharToShortTimestampCoercer
+            extends TypeCoercer<VarcharType, TimestampType>
+    {
+        public VarcharToShortTimestampCoercer(VarcharType fromType, TimestampType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            try {
+                Slice value = fromType.getSlice(block, position);
+                LocalDateTime dateTime = LocalDateTime.from(LOCAL_DATE_TIME.parse(value.toStringUtf8()));
+                toType.writeLong(
+                        blockBuilder,
+                        (dateTime.toEpochSecond(UTC) * MICROSECONDS_PER_SECOND) + floorDiv(dateTime.getNano(), NANOSECONDS_PER_MICROSECOND));
+            }
+            catch (DateTimeParseException exception) {
```

Comment:
Should we log this?

Reply:
I'm not sure if the table is pretty huge like 1B and if all of them doesn't support then it - it might be a huge log (even if we set them as a DEBUG level)

Reply:
add a comment why you append null instead of eg propagating exception